src/services/ui/src/api/useGetTypes.ts

```diff
@@ -0,0 +1,32 @@
+import { API } from "aws-amplify";
+import { useQuery, UseQueryOptions } from "@tanstack/react-query";
+import { opensearch, ReactQueryApiError } from "shared-types";
+import { types } from "shared-types/opensearch";
+
+async function fetchTypes(
+  authorityId: number
+): Promise<opensearch.types.Document[]> {
+  const endpoint = "/getTypes";
+  const body = { authorityId };
+
+  try {
+    const response = await API.post("os", endpoint, { body });
+    const hits = response.hits?.hits || [];
+
+    return hits.map((hit: types.ItemResult) => hit._source);
+  } catch (error) {
+    console.error("Error fetching types:", error);
+    throw new Error("Failed to fetch types");
+  }
+}
```

Comment:
This chunk here is testable. Let me know if you want to huddle and go over setting up an `msw` interceptor to test it